contrib/jupyter/build.py

```diff
@@ -5,24 +5,39 @@
 """
 
 import os, sys
-import orbitutils.run_all_unittests
+import build
+import subprocess
 
-file_directory = os.path.dirname(os.path.abspath(__file__))
-orbitutils_path = os.path.join(file_directory, "orbitutils")
 
-# We need to specify the "root" of the src tree. With this choice, we can then
-# import the proto as "import ClientProtos.capture_data_pb2.py"
-proto_path = os.path.join(file_directory, "../../src/")
+def install_orbitutils(current_directory):
+    subprocess.check_call(
+        [sys.executable, "-m", "pip", "install", "-e", current_directory])
 
-# This is the only .proto file we currently need for our python code.
-capture_data_proto = os.path.join(file_directory,
-                                  "../../src/ClientProtos/capture_data.proto")
 
-command = "protoc --python_out=" + orbitutils_path + " --proto_path=" + proto_path + " " + capture_data_proto
-ret = os.system(command)
-if ret:
-    print("Command failed: ", command)
-    sys.exit(ret)
+def main():
+    current_directory = os.path.dirname(os.path.abspath(__file__))
+    orbitutils_path = os.path.join(current_directory, "orbitutils")
 
-# Verify by running all unit tests
-orbitutils.run_all_unittests.main()
\ No newline at end of file
+    # We need to specify the "root" of the src tree. With this choice, we can then
+    # import the proto as "import ClientProtos.capture_data_pb2.py"
+    proto_path = os.path.join(current_directory, "../../src/")
+
+    # This is the only .proto file we currently need for our python code.
+    capture_data_proto = os.path.join(
+        current_directory, "../../src/ClientProtos/capture_data.proto")
+
+    command = "protoc --python_out=" + orbitutils_path + " --proto_path=" + proto_path + " " + capture_data_proto
+    ret = os.system(command)
+    if ret:
+        print("Command failed: ", command)
+        sys.exit(ret)
+
+    install_orbitutils(current_directory)
+
+    subprocess.check_call(
```

Comment:
Calling this as a subprocess ensures that orbitutils is installed when executing the imports in run_tests.py. There are instructions how to do an "try import; install if failed; retry import" on https://stackoverflow.com/questions/12332975/installing-python-module-within-code, but this did not work for me. So, right now I don't know if there is a more elegant solution. I might just be doing something wrong.

Reply:
I think the subprocess approach is viable. I consider it coincidence that the language we use for writing this script is the same language as the code under test. If this was a bash script this would be totally fine.

FWIW, I think you might need to reload your import path:
```
import site
from importlib import reload
reload(site)
```
Ref: https://stackoverflow.com/questions/25384922/how-to-refresh-sys-path

Reply:
Hmm, I couldn't figure out how to do the reload in a few minutes (it also seemed to have changed with Python 3), so I think I'll just go with the subprocess approach.